python/ray/experimental/serve/task_runner.py

```diff
@@ -121,8 +126,75 @@ def _ray_serve_call(self, request):
                                                 result_object_id)
         self._serve_metric_latency_list.append(time.time() - start_timestamp)
 
-        serve_context.web = False
+    def invoke_batch(self, request_item_list):
+        # TODO(alind) : create no-http services. The enqueues
+        # from such services will always be TaskContext.Python.
+
+        # Assumption : all the requests in a bacth
+        # have same serve context.
+
+        # For batching kwargs are modified as follows -
+        # kwargs [Python Context] : key,val
+        # kwargs_list             : key, [val1,val2, ... , valn]
+        # or
+        # args[Web Context]       : val
+        # args_list               : [val1,val2, ...... , valn]
+        # where n (current batch size) <= max_batch_size of a backend
+
+        kwargs_list = defaultdict(list)
+        result_object_ids, context_list, arg_list = [], [], []
+        for item in request_item_list:
+            args, kwargs, web_context, r_object_id = self._parse_request_item(
+                item)
+            context_list.append(web_context)
+            if not web_context:
+                for k, v in kwargs.items():
+                    kwargs_list[k].append(v)
+            else:
+                arg_list.append(args[0])
+
+            result_object_ids.append(r_object_id)
+        try:
+            # check mixing of query context
+            if not (all(context_list) or not any(context_list)):
```

Comment:
`if set(context_list) != 1` is a bit clearer saying that we want unified context.